math-scala/src/main/scala/org/apache/mahout/math/drm/DistributedEngine.scala

```diff
@@ -119,38 +140,49 @@ object DistributedEngine {
   private def pass1[K: ClassTag](action: DrmLike[K]): DrmLike[K] = {
 
     action match {
-      case OpAB(OpAt(a), b) if (a == b) => OpAtA(pass1(a))
-      case OpABAnyKey(OpAtAnyKey(a), b) if (a == b) => OpAtA(pass1(a))
+
+      // self element-wise rewrite
+      case OpAewB(a, b, op) if (a == b) => {
```

Comment:
I am using the unicode chars now (change of style that happened to me a few months ago) but it has not gone over the entire codebase. We probably may want to standardize on this (later) but for now i just keep style consistent with the rest of the file. If i change the style, i change the style in the whole file.